### PR TITLE
Add ViT Bart Conditional Generation Model

### PR DESCRIPTION
**Pull request by gchhablani**
**Note**: This model doesn't support `.generate()` just yet. Needs fixing.

### CODE REVIEW

File: models/flax_vit_bart/modeling_vit_bart.py [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+            deterministic=not train,
+            rngs=rngs,
+        )
+
+
+class FlaxViTBartForConditionalGeneration(FlaxViTBartPreTrainedModel):
+    module_class = FlaxViTBartForConditionalGenerationModule
+    dtype: jnp.dtype = jnp.float32
+
+    def decode(
```
Comment (patil-suraj) [inline models/flax_vit_bart/modeling_vit_bart.py:490]: Note that it's not required to create this class, if you just have one downstream model, you could just set 
rename `FlaxViTBartPreTrainedModel` to `FlaxViTBartForConditionalGenerationModule` and set the
`module_class = FlaxViTBartForConditionalGenerationModule`, and also add the necessary methods like `prepare_inputs_for_generation` then that should also work.

(you could keep it as it is, but commenting here to make clear).

**Review by patil-suraj — COMMENTED** [review]
Really great work !
This looks good to me!

When you are training remember to shift the labels to the right to get the correct `decoder_input_ids`, as is done here https://github.com/huggingface/transformers/blob/master/examples/flax/summarization/run_summarization_flax.py#L453

File: models/flax_vit_bart/modeling_vit_bart.py (hunk 1 above)
Comment (gchhablani) [reply models/flax_vit_bart/modeling_vit_bart.py:490]: Thanks @patil-suraj for taking the time and reviewing :)
